tests/packages/Microsoft.Bot.Components.Telephony.Tests/Integration/CallTransferTests/CallTransfer_BaseScenario.test.dialog

```diff
@@ -5,12 +5,16 @@
     "triggers": [
         {
             "$kind": "Microsoft.OnBeginDialog",
-            "actions": [
-                {
-                    "$kind": "Microsoft.Telephony.CallTransfer",
-                    "phoneNumber": "+1555443443"
-                }
-            ]
+          "actions": [
+            {
+              "$kind": "Microsoft.Telephony.CallTransfer",
+              "phoneNumber": "+1555443443"
```

Comment:
Regding negative test for invalid or missing number: phoneNumber needs to be marked as mandatory or validated in the action. 

    var phoneNumber = this.PhoneNumber?.GetValue(dc.State); // accepts empty and invalid numbers

Reply:
I have added validation to the phone number

Reply:
I understood from the chat that we'd validate non-null but we'd leave specific validation to the channels so we don't hardcode provider specific validation. I also understood that if we pass a non-null value along with the event, we'd get an error from the channel if the number is not valid. Are we saying that we'll end up validating 1) non-null and 2) phone regex  in the action then?

Regarding getting the error back on bad phone numbers, which was discussed yesterday, have you tested it with the actual telephony channel? How do we get the error back? do we get it as response to the event activity or how? We should make sure that we have a solid story here in capturing such errors.

Reply:
Please look at my latest PR for the error 

var error = "Phone number is empty, please provide number in E.164 format";
 return await dc.EndDialogAsync(result: error, cancellationToken: cancellationToken).ConfigureAwait(false);

this error will be caught in runtime, when actual phone call will be made via telephony and call transfer is initiated, so not sure where we should bubble this up per say.

Reply:
I see that you added the error, my comment was about whether to perform validation in the bot side or in the channel / provider side as we discussed in the crew. Let's talk this in a call.

Reply:
yes, I spoke to TC team and they dont do validation of phonenumber on their side, they rely on validation on our side else it will fail in the lowest level, which could be detected much earlier by us, hence I am adding validation here.

Reply:
Thanks! Let's discuss this with Pooja on Tuesday.

Reply:
discussed with Pooja offline, we will not do validation for format. But we need your input on how to localize error for null case